cfnet/datasets.py

```diff
@@ -32,23 +31,52 @@ def _numpy_collate(batch):
   else:
     return np.array(batch)
 
-class NumpyLoader(DataLoader):
-  def __init__(self, dataset, batch_size=1,
+class DataLoader:
+    def __init__(self,dataset, batch_size=1,
                 shuffle=False, sampler=None,
-                batch_sampler=None, num_workers=0,
+                batch_sampler=None, num_workers=0,collate_fn=_numpy_collate,
                 pin_memory=False, drop_last=False,
                 timeout=0, worker_init_fn=None):
-    super(self.__class__, self).__init__(dataset,
-        batch_size=batch_size,
-        shuffle=shuffle,
-        sampler=sampler,
-        batch_sampler=batch_sampler,
-        num_workers=num_workers,
-        collate_fn=_numpy_collate,
-        pin_memory=pin_memory,
-        drop_last=drop_last,
-        timeout=timeout,
-        worker_init_fn=worker_init_fn)
+        # Attributes from pytorch data loader
+        self.dataset = dataset
+        self.batch_size = batch_size
+        self.shuffle=shuffle
+        self.sampler=sampler
+        self.batch_sampler=batch_sampler
+        self.num_workers=num_workers
+        self.collate_fn=collate_fn
+        self.pin_memory=pin_memory
+        self.drop_last=drop_last
+        self.timeout=timeout
+        self.worker_init_fn=worker_init_fn
+
+        self.dataLen = len(dataset)
+        self.keySeq = hk.PRNGSequence(42)
+        self.keySeq.reserve(len(self))
+        self.key = next(self.keySeq)
+        self.indices = jax.numpy.arange(self.dataLen)
+        self.pose = 0
+    def __len__(self):
+        batches = -(len(self.dataset) // -self.batch_size)
```

Comment:
Why are you using `-`. If not necessary, please remove them.

Reply:
Use - to get the ceil instead of floor.

Reply:
Give a brief comment of this line